packages/tests/tests/peer_exchange.node.spec.ts

```diff
@@ -35,56 +36,31 @@ describe("Peer Exchange", () => {
     waku = await createLightNode({
       libp2p: {
         peerDiscovery: [
-          bootstrap({ list: getPredefinedBootstrapNodes(Fleet.Test) }),
+          bootstrap({ list: getPredefinedBootstrapNodes(Fleet.Test, 3) }),
           wakuPeerExchangeDiscovery(),
         ],
       },
     });
 
     await waku.start();
-    // we want to ensure that there is enough time for discv5 to discover peers
-    await delay(40000);
-
-    await waitForRemotePeer(waku, [Protocols.PeerExchange]);
-    const pxPeers = await waku.peerExchange.peers();
-    expect(pxPeers.length).to.be.greaterThan(0);
-  });
 
-  it("Manual query on test fleet", async function () {
-    this.timeout(60_000);
-
-    const waku = await createLightNode({
-      libp2p: {
-        peerDiscovery: [
-          bootstrap({ list: getPredefinedBootstrapNodes(Fleet.Test) }),
-        ],
-      },
+    let foundPxPeer = false;
+
+    await new Promise((resolve) => {
+      const testNodes = getPredefinedBootstrapNodes(Fleet.Test, 3);
+      waku.libp2p.addEventListener("peer:discovery", (evt) => {
+        const { multiaddrs } = evt.detail;
+        multiaddrs.forEach((ma) => {
+          const isBootstrapNode = testNodes.find((n) => n === ma.toString());
+          if (!isBootstrapNode) {
+            foundPxPeer = true;
+            resolve(evt.detail.id);
```

Comment:
You don't use the result of the promise. Just do `resolve(true)` and assign the result of the promise to `foundPxPeer`